make/autoconf/jvm-features.m4

```diff
@@ -47,9 +47,9 @@
 m4_define(jvm_features_valid, m4_normalize( \
     ifdef([custom_jvm_features_valid], custom_jvm_features_valid) \
     \
-    cds compiler1 compiler2 dtrace epsilongc g1gc jfr jni-check \
-    jvmci jvmti link-time-opt management minimal opt-size parallelgc \
-    serialgc services shenandoahgc static-build vm-structs zero zgc \
+    dtrace jfr \
+    link-time-opt management opt-size \
+    static-build \
```

Comment:
Do we want to remove `static-build` from this list as well? It's only available on macOS, but I don't know if it works with OpenJ9.

Suggest the explicit `JVM_FEATURES_CHECK_xxx` functions be updated as well to avoid misleading output from the configuration step: for example, it should not say
```
checking if platform is supported by CDS... yes
checking if JVM feature 'cds' is available... yes
```
instead, `JVM_FEATURES_CHECK_CDS` should be updated, like `JVM_FEATURES_CHECK_JVMCI` so it says
```
checking if platform is supported by CDS... no, OpenJ9
checking if JVM feature 'cds' is available... no
```
Similar changes should be made to `JVM_FEATURES_CHECK_SHENANDOAHGC` and `JVM_FEATURES_CHECK_ZGC`.

Now that CDS is disabled here, lines 682-684 in `custom-hook.m4` are redundant.

Reply:
Updated.